src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -585,6 +585,27 @@ def set_deployment_user(user_name, password=None):
     result = client.update_publishing_user(user)
     return result
 
+def list_publish_profiles(resource_group_name, name, slot=None):
+    from azure.mgmt.web.models import PublishingProfileFormat
+    import xmltodict
+
+    content = _generic_site_operation(resource_group_name, name,
+                                      'list_publishing_profile_xml_with_secrets', slot)
+    full_xml = ''
+    for f in content:
+        full_xml += f.decode()
+
+    profiles = xmltodict.parse(full_xml, xml_attribs=True)['publishData']['publishProfile']
```

Comment:
It should always be there. Since I worked on it from Visual Studio Tooling side several years ago, I have never seen the opposite :). So i suggest we keep the assumption. The worst case is we see a bad error and we should figure out why either way at that time.